python/port/mod/turtle/turtle.cpp

```diff
@@ -0,0 +1,239 @@
+#include "turtle.h"
+extern "C" {
+#include <py/misc.h>
+}
+#include "../../helpers.h"
+#include "../../port.h"
+#include "turtle_icon.h"
+
+static constexpr KDSize k_iconSize = KDSize(9, 9);
+
+template <typename T> static inline T * allocate(size_t count) {
+  /* We forward dynamic allocations to the Python GC so we don't have to bother
+   * with deallocation. For this to work well, the Turtle object who owns the
+   * allocated areas need to be added to MicroPython's GC roots, otherwise those
+   * buffers will be wiped out by the GC too early. */
+  return static_cast<T*>(m_malloc(sizeof(T) * count));
```

Comment:
What happens if MicroPython run out of heap?

Reply:
Well, literally nothing! Dynamic memory is used for drawing operations. If a given allocation fails, the matching operations are skipped!